CreditManagement/models.py

```diff
@@ -438,14 +492,43 @@ class UserCredit(models.Model):
     user = models.OneToOneField(User, primary_key=True,
                                   db_column='id',
                                 on_delete=models.DO_NOTHING)
-    balance = models.DecimalField(blank=True, null=True, db_column='balance', decimal_places=2, max_digits=6)
+    balance = models.DecimalField(blank=True, null=True,
+                                  db_column=AbstractTransaction.balance_annotation_name,
+                                  decimal_places=2,
+                                  max_digits=6)
+    balance_fixed = models.DecimalField(blank=True,
+                                        null=True,
+                                        db_column=FixedTransaction.balance_annotation_name,
+                                        decimal_places=2,
+                                        max_digits=6)
+
+    def negative_since(self):
+        """
+        Compute the date in the confirmed moments that
+        :return:
+        """
+        balance = self.balance_fixed
+        if balance >= 0:
+            # balance is already positive, return nothing
+            return None
+
+        # Loop over all transactions from new to old, reverse its balance
+        transactions = FixedTransaction.get_all_transactions(user=self.user).order_by('-order_moment')
+        for transaction in transactions:
+            balance += transaction.amount
+            # If balance is positive now, return the current transaction date
+            if balance >= 0:
+                return transaction.order_moment
+
+        # This should not be reached, it would indicate that the starting balance was 0
+        raise Exception("Balance started as negative, negative_since could not be computed")
```

Comment:
Ideally you would make a custom Exception subclass for this

Reply:
Now changed into RunTimeError